
JB: I think these declarations should only be in the corresponding driver...

Index: linux-2.6.29/arch/arm/mach-imx/include/mach/imx-regs.h
===================================================================
--- linux-2.6.29.orig/arch/arm/mach-imx/include/mach/imx-regs.h	2009-05-01 11:33:45.000000000 +0200
+++ linux-2.6.29/arch/arm/mach-imx/include/mach/imx-regs.h	2009-05-01 14:02:48.000000000 +0200
@@ -373,4 +373,40 @@
 #define TSTAT_CAPT         (1<<1)
 #define TSTAT_COMP         (1<<0)
 
+/*
+ * CSI register
+ */
+#define  CSICR1        __REG(0x00 + IMX_CSI_BASE)
+#define  CSICR2        __REG(0x04 + IMX_CSI_BASE)
+#define  CSISR         __REG(0x08 + IMX_CSI_BASE)
+#define  CSISTATR      __REG(0x0C + IMX_CSI_BASE)
+#define  CSIRXR        __REG(0x10 + IMX_CSI_BASE)
+
+/* CSI bit fields */
+#define CSICR1_RXOR_INTEN      (1<<24)
+#define CSICR1_RXFF_INTEN      (1<<18)
+#define CSICR1_RXFF_LEVEL(x)   ((x & 0x3)<<19)
+#define CSICR1_SOF_POL         (1<<17)
+#define CSICR1_SOF_INTEN       (1<<16)
+#define CSICR1_MCLKDIV(x)      ((x & 0xf)<<12)
+#define CSICR1_MCLKEN          (1<<9)
+#define CSICR1_FCC             (1<<8)
+#define CSICR1_BIG_ENDIAN      (1<<7)
+#define CSICR1_CLR_STATFIFO    (1<<6)
+#define CSICR1_CLR_RXFIFO      (1<<5)
+#define CSICR1_GCLK_MODE       (1<<4)
+#define CSICR1_INV_DATA        (1<<3)
+#define CSICR1_REDGE           (1<<1)
+#define CSICR1_ENABLE          (1)
+
+/* SPI regiters */
+#define SPI1_RXDATA        __REG(IMX_SPI1_BASE + 0x00)
+#define SPI1_TXDATA        __REG(IMX_SPI1_BASE + 0x04)
+#define SPI1_CONTROL       __REG(IMX_SPI1_BASE + 0x08)
+#define SPI1_INT           __REG(IMX_SPI1_BASE + 0x0c)
+#define SPI1_TEST          __REG(IMX_SPI1_BASE + 0x10)
+#define SPI1_PERIOD        __REG(IMX_SPI1_BASE + 0x14)
+#define SPI1_DMA           __REG(IMX_SPI1_BASE + 0x18)
+#define SPI1_RESET         __REG(IMX_SPI1_BASE + 0x1c)
+
 #endif				// _IMX_REGS_H
